src/main/java/com/github/jparkie/pdd/BitArray.java

```diff
@@ -38,22 +38,35 @@ private static int numWords(long numBits) {
         return (int) numWords;
     }
 
+    private final class indexAndMask {
+        public final int index;
+        public final long mask;
+
+        public indexAndMask(long index) {
+            this.index = (int)(index >>> 6);
+            this.mask = 1L << (index & 0x3F);
+        }
+    }
+
     public boolean get(long index) {
-        return (data[(int) (index >>> 6)] & (1L << index)) != 0;
+        final indexAndMask iam = new indexAndMask(index);
```

Comment:
Yep, GC oressure was a worry. As a CLR/JIT/C# hack, I was just trying to emulate what I would do in C#
`private void computeIndexAndMask(long index, out int arrayIndex, out long bitMask) {` and this was as close as I could get without expending time.  In the CLR world, this would have been a `struct` and would have been stack-allocated and inlined it registers by the JIT.
I'm learning.